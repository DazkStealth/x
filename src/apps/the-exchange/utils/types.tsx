import { BridgingProvider, ExchangeOffer } from '@etherspot/prime-sdk/dist/sdk/data';
import { Route } from '@lifi/sdk';

export enum CardPosition {
    SWAP = 'SWAP',
    RECEIVE = 'RECEIVE'
}

export type SwapType = {
    fromAmount: number,
    fromTokenAddress: string,
    fromChainId: number;
    fromTokenDecimals: number;
    toTokenAddress: string,
    toChainId: number,
    toTokenDecimals: number;
    slippage: number;
    fromAccountAddress?: string
    provider?: BridgingProvider;
}

export interface SwapOffer {
    tokenAmountToReceive: number;
    offer: Route | ExchangeOffer;
}
