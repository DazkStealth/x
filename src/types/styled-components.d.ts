import 'styled-components';

// theme
import { Theme } from '../theme';

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
