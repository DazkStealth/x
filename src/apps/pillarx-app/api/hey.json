{
    "projection": [
      {
        "meta": {},
        "data": {
          "total_wallet_balance": 3601.2644252510418,
          "wallet": "0x11041744893fa72629ab93ea8adaf35a1dc24aa5",
          "wallets": [
            "0x11041744893fa72629ab93ea8adaf35a1dc24aa5"
          ],
          "total_realized_pnl": 1657.3989,
          "total_unrealized_pnl": 683.2397000000001,
          "assets": [
            {
              "asset": {
                "name": "Arbitrum",
                "symbol": "ARB",
                "id": 511,
                "contracts": [
                  "0x912ce59144191c1204e64559fe8253a0e49e6548",
                  "0xb50721bcf8d664c30412cfbc6cf7a15145234ad1",
                  "0xf823c3cd3cebe0a1fa952ba88dc9eef8e0bf46ad"
                ],
                "logo": "https://metacore.mobula.io/55419fdf33d8a53e61b0bd55aa6a94a855c34a48cb62ca4766e5ac89a4e90887.png",
                "blockchains": [
                  "Arbitrum",
                  "Ethereum",
                  "Arbitrum Nova"
                ]
              },
              "realized_pnl": 102.3196,
              "unrealized_pnl": -95.9317,
              "allocation": 19.5,
              "price": 1.194947271373868,
              "price_bought": 0,
              "price_change_24h": -3.241148,
              "price_change_1h": 0,
              "total_invested": 0,
              "min_buy_price": 1.3450628571428571,
              "max_buy_price": 1.5500332634003993,
              "estimated_balance": 702.2078712586506,
              "token_balance": 587.64757917,
              "cross_chain_balances": {
                "Arbitrum": {
                  "balance": 587.64757917,
                  "balanceRaw": "587647579169999945728",
                  "chainId": "42161",
                  "address": "0x912ce59144191c1204e64559fe8253a0e49e6548"
                }
              },
              "contracts_balances": [
                {
                  "balance": 587.64757917,
                  "balanceRaw": "587647579169999945728",
                  "chainId": "42161",
                  "address": "0x912ce59144191c1204e64559fe8253a0e49e6548",
                  "decimals": 18
                }
              ]
            },
      
            {
              "asset": {
                "name": "Orbit Protocol",
                "symbol": "ORBIT",
                "id": 102480544,
                "contracts": [
                  "0x42e12d42b3d6c4a74a88a61063856756ea2db357"
                ],
                "logo": "https://metacore.mobula.io/958c9264cc15c3fdf2d6ca544010be9d8385d373d7e441c6a9239e07af301e8c.png",
                "blockchains": [
                  "Blast"
                ]
              },
              "realized_pnl": 0,
              "unrealized_pnl": -29.3324,
              "allocation": 0.15,
              "price": 0.1383163581556911,
              "price_bought": 0,
              "price_change_24h": -14.07113,
              "price_change_1h": 0,
              "total_invested": 0,
              "min_buy_price": 0.8745584448725503,
              "max_buy_price": 0.8745584448725503,
              "estimated_balance": 5.510617266881962,
              "token_balance": 39.8406764056,
              "cross_chain_balances": {
                "Blast": {
                  "balance": 39.84067640565484,
                  "balanceRaw": "39840676405654839296",
                  "chainId": "81457",
                  "address": "0x42e12d42b3d6c4a74a88a61063856756ea2db357"
                }
              },
              "contracts_balances": [
                {
                  "balance": 39.84067640565484,
                  "balanceRaw": "39840676405654839296",
                  "chainId": "81457",
                  "address": "0x42e12d42b3d6c4a74a88a61063856756ea2db357",
                  "decimals": 18
                }
              ]
            }
          ],
          "total_pnl_history": {
            "24h": {
              "realized": 0,
              "unrealized": -114.0661999999999
            },
            "7d": {
              "realized": 0,
              "unrealized": -169.20839999999998
            },
            "30d": {
              "realized": 74.44840000000019,
              "unrealized": 91.57700000000004
            },
            "1y": {
              "realized": 350.1728000000002,
              "unrealized": 955.3388000000001
            }
          }
        },
        "layout": "LIST_HORIZONTAL",
        "id": "walletPortfolio"
      }
    ]
  }