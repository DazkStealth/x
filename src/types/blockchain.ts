import { BigNumberish } from 'ethers';

export interface Transaction {
  id: string;
  value: BigNumberish;
  to: string;
  data?: string;
  status: string;
  hash?: string;
  userOpHash?: string;
}
