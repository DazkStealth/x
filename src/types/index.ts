export interface AppManifest {
  title: string;
  description: string;
  translations: RecordPerKey<string>;
}

export interface RecordPerKey<T> {
  [key: string]: T;
}
