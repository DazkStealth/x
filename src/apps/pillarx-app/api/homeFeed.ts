/* eslint-disable no-console */
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { addMiddleware } from '../../../store';
import { ApiResponse } from '../../../types/api';

export const homeFeedApi = createApi({
    reducerPath: 'homeFeedApi',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://feed-nubpgwxpiq-uc.a.run.app/' }),
    endpoints: (builder) => ({
        getTilesInfo: builder.query<ApiResponse, { page: number; address: string }>({
            query: ({ page, address }) => `?page=${page}&address=${address}`,
        }),
    })
})

/**
 * Add this to the store
 */
addMiddleware(homeFeedApi);

export const { useGetTilesInfoQuery } = homeFeedApi;
