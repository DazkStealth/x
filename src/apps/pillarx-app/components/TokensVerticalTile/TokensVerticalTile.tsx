// types
import { TokenData } from '../../../../types/api';

// components
import TileContainer from '../TileContainer/TileContainer'
import SkeletonLoader from '../../../../components/SkeletonLoader';

import TokensVerticalList from '../TokensVerticalList/TokensVerticalList';

type TokensVerticalTileProps = {
    dataLeft: TokenData[] | undefined;
    dataRight: TokenData[] | undefined;
    titleLeft: string;
    titleRight: string;
    isDataLoading: boolean;
}

const TokensVerticalTile = ({ dataLeft, dataRight, titleLeft, titleRight, isDataLoading }: TokensVerticalTileProps) => {

if (!dataLeft || !dataRight || isDataLoading) {
    return (
    <TileContainer className='px-10 pt-[30px] pb-5 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col'>
         <div className='flex flex-col flex-1 mr-5 mobile:m-0'>
            <SkeletonLoader  $height='24px' $width='120px' $marginBottom='20px' $radius='10px' />
            <div className='flex flex-col'>
                <SkeletonLoader  $height='50px' $width='100%' $marginBottom='20px' $radius='10px' />
                <SkeletonLoader  $height='50px' $width='100%' $marginBottom='20px' $radius='10px' />
                <SkeletonLoader  $height='50px' $width='100%' $marginBottom='20px' $radius='10px' />
            </div>
        </div>
        <div className='flex flex-col flex-1 ml-5 mobile:m-0'>
            <SkeletonLoader  $height='24px' $width='120px' $marginBottom='20px' $radius='10px' />
            <div className='flex flex-col'>
                <SkeletonLoader  $height='50px' $width='100%' $marginBottom='20px' $radius='10px' />
                <SkeletonLoader  $height='50px' $width='100%' $marginBottom='20px' $radius='10px' />
                <SkeletonLoader  $height='50px' $width='100%' $marginBottom='20px' $radius='10px' />
            </div>
        </div>
    </TileContainer>
    )
}

    return (
    <TileContainer className='px-10 pt-[30px] pb-5 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col'>
        <TokensVerticalList position='left' data={dataLeft} title={titleLeft} />
        <TokensVerticalList position='right' data={dataRight} title={titleRight} />
    </TileContainer>
    )
};

export default TokensVerticalTile;
