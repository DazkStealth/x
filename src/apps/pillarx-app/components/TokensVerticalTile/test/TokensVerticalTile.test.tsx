import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import TokensVerticalTile from '../TokensVerticalTile';

// types
import { TokenData } from '../../../../../types/api';

describe('<TokensVerticalTile />', () => {
  const mockDataLeft: TokenData[] = [
        {
          id: 1,
          name: 'Token A', 
          symbol: 'TKA',
          contracts: [{ address: '0x1', blockchain: 'Ethereum' }],
          logo: 'logoA.png',
          trending_score: 90,
          platforms: [{ name: 'Platform A', rank: 1, weight: 0.5 }],
          price_change_24h: 5,
          pair: 'TKA/USD'
        },
        {
          id: 2,
          name: 'Token B', 
          symbol: 'TKB',
          contracts: [{ address: '0x2', blockchain: 'Ethereum' }],
          logo: 'logoB.png',
          trending_score: 80,
          platforms: [{ name: 'Platform B', rank: 2, weight: 0.4 }],
          price_change_24h: -3,
          pair: 'TKB/USD' 
        }
      ];

   const mockDataRight: TokenData[] = [
    {
      id: 1,
      name: 'Token A', 
      symbol: 'TKA',
      contracts: [{ address: '0x1', blockchain: 'Ethereum' }],
      logo: 'logoA.png',
      trending_score: 90,
      platforms: [{ name: 'Platform A', rank: 1, weight: 0.5 }],
      price_change_24h: 5,
      pair: 'TKA/USD'
    },
    {
      id: 2,
      name: 'Token B', 
      symbol: 'TKB',
      contracts: [{ address: '0x2', blockchain: 'Ethereum' }],
      logo: 'logoB.png',
      trending_score: 80,
      platforms: [{ name: 'Platform B', rank: 2, weight: 0.4 }],
      price_change_24h: -3,
      pair: 'TKB/USD' 
    }
  ];

    it('renders correctly and matches snapshot', () => {
      const tree = renderer
          .create(<TokensVerticalTile dataLeft={mockDataLeft} dataRight={mockDataRight} titleLeft='Top climbers' titleRight='Top losers' isDataLoading={false} />)
          .toJSON();

      expect(tree).toMatchSnapshot();
    });


    it('displays loading skeleton when data is loading', () => {
        const tree = renderer
            .create(<TokensVerticalTile dataLeft={undefined} dataRight={undefined} titleLeft='' titleRight='' isDataLoading={true} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with empty data array', () => {
      const tree = renderer
        .create(<TokensVerticalTile dataLeft={[]} dataRight={[]} titleLeft='' titleRight='' isDataLoading={false} />)
        .toJSON();
  
      expect(tree).toMatchSnapshot();
    });
  
    it('renders TokensVerticalList components with correct titles', () => {
      const tree = renderer.create(<TokensVerticalTile dataLeft={mockDataLeft} dataRight={mockDataRight} titleLeft='Top climbers' titleRight='Top losers' isDataLoading={false} />).toJSON() as ReactTestRendererJSON;
      const verticalLists = tree.children as ReactTestRendererJSON[];

      const leftTitleProp = (verticalLists[0].children?.find(child => 
        typeof child === 'object' && child.type === 'p') as ReactTestRendererJSON);

      const rightTitleProp = (verticalLists[1].children?.find(child => 
        typeof child === 'object' && child.type === 'p') as ReactTestRendererJSON);
  
      expect(leftTitleProp.children).toContain('Top climbers');
      expect(rightTitleProp.children).toContain('Top losers');
    });
});
