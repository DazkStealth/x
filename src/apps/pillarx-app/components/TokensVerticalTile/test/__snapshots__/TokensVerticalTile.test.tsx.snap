// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokensVerticalTile /> displays loading skeleton when data is loading 1`] = `
.c0 {
  height: 24px;
  width: 120px;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 10px;
  margin-bottom: 20px;
}

.c1 {
  height: 50px;
  width: 100%;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 10px;
  margin-bottom: 20px;
}

<div
  className="flex bg-container_grey rounded-2xl px-10 pt-[30px] pb-5 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col"
>
  <div
    className="flex flex-col flex-1 mr-5 mobile:m-0"
  >
    <div
      className="c0"
    />
    <div
      className="flex flex-col"
    >
      <div
        className="c1"
      />
      <div
        className="c1"
      />
      <div
        className="c1"
      />
    </div>
  </div>
  <div
    className="flex flex-col flex-1 ml-5 mobile:m-0"
  >
    <div
      className="c0"
    />
    <div
      className="flex flex-col"
    >
      <div
        className="c1"
      />
      <div
        className="c1"
      />
      <div
        className="c1"
      />
    </div>
  </div>
</div>
`;

exports[`<TokensVerticalTile /> renders correctly and matches snapshot 1`] = `
<div
  className="flex bg-container_grey rounded-2xl px-10 pt-[30px] pb-5 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col"
>
  <div
    className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pr-10 border-r-[3px] border-[#1F1D23]"
  >
    <p
      className="text-base font-medium text-purple_light"
    >
      Top climbers
    </p>
    <div
      className="flex flex-col"
    >
      <div
        className="flex w-full justify-between py-5 border-b border-[#1F1D23] mobile:border-[#27262F]"
      >
        <div
          className="flex items-center"
        >
          <p
            className="text-sm font-medium text-purple_light mr-[18px]"
          >
            0
            1
          </p>
          <img
            className="w-[50px] h-[50px] object-fill rounded mr-3.5"
            src="logo-unknown.png"
          />
          <div
            className="flex flex-col"
          >
            <p
              className="text-base font-medium undefined"
            >
              TKA
            </p>
            <p
              className="text-sm font-medium text-purple_light"
            >
              Token A
            </p>
          </div>
        </div>
        <div
          className="flex flex-col items-center"
        />
      </div>
      <div
        className="flex w-full justify-between py-5 border-b border-[#1F1D23] mobile:border-[#27262F]"
      >
        <div
          className="flex items-center"
        >
          <p
            className="text-sm font-medium text-purple_light mr-[18px]"
          >
            0
            2
          </p>
          <img
            className="w-[50px] h-[50px] object-fill rounded mr-3.5"
            src="logo-unknown.png"
          />
          <div
            className="flex flex-col"
          >
            <p
              className="text-base font-medium undefined"
            >
              TKB
            </p>
            <p
              className="text-sm font-medium text-purple_light"
            >
              Token B
            </p>
          </div>
        </div>
        <div
          className="flex flex-col items-center"
        />
      </div>
    </div>
  </div>
  <div
    className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pl-10"
  >
    <p
      className="text-base font-medium text-purple_light"
    >
      Top losers
    </p>
    <div
      className="flex flex-col"
    >
      <div
        className="flex w-full justify-between py-5 border-b border-[#1F1D23] mobile:border-[#27262F]"
      >
        <div
          className="flex items-center"
        >
          <p
            className="text-sm font-medium text-purple_light mr-[18px]"
          >
            0
            1
          </p>
          <img
            className="w-[50px] h-[50px] object-fill rounded mr-3.5"
            src="logo-unknown.png"
          />
          <div
            className="flex flex-col"
          >
            <p
              className="text-base font-medium undefined"
            >
              TKA
            </p>
            <p
              className="text-sm font-medium text-purple_light"
            >
              Token A
            </p>
          </div>
        </div>
        <div
          className="flex flex-col items-center"
        />
      </div>
      <div
        className="flex w-full justify-between py-5 border-b border-[#1F1D23] mobile:border-[#27262F]"
      >
        <div
          className="flex items-center"
        >
          <p
            className="text-sm font-medium text-purple_light mr-[18px]"
          >
            0
            2
          </p>
          <img
            className="w-[50px] h-[50px] object-fill rounded mr-3.5"
            src="logo-unknown.png"
          />
          <div
            className="flex flex-col"
          >
            <p
              className="text-base font-medium undefined"
            >
              TKB
            </p>
            <p
              className="text-sm font-medium text-purple_light"
            >
              Token B
            </p>
          </div>
        </div>
        <div
          className="flex flex-col items-center"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<TokensVerticalTile /> renders correctly with empty data array 1`] = `
<div
  className="flex bg-container_grey rounded-2xl px-10 pt-[30px] pb-5 tablet:p-5 mobile:p-0 mobile:bg-[#1F1D23] mobile:flex-col"
>
  <div
    className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pr-10 border-r-[3px] border-[#1F1D23]"
  >
    <div
      className="flex flex-col"
    />
  </div>
  <div
    className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pl-10"
  >
    <div
      className="flex flex-col"
    />
  </div>
</div>
`;
